### ISSUE TITLE
RectangularRadioButtonGroup highlights need to update with dynamic locales

### ISSUE THREAD

**Issue by jessegreenberg** [phetsims/sun#851]
Identified in https://github.com/phetsims/greenhouse-effect/issues/349, the radio button groups are not adjusting with dynamic layout. This issue is specifically for the highlights, which aren't updating because the default is overridden. It happens here, so it is likely an issue for mouse/touch areas too.

https://github.com/phetsims/sun/blob/f0b67964a1beedb065ae7f6839943eebaa11f26a/js/buttons/RectangularRadioButtonGroup.ts#L278-L304

**Comment by jbphet** [comment]
@jessegreenberg - I started out reviewing this by checking the functionality, and I can see that it's now better for the wider button, but the highlights on the other buttons don't look correct.  See the screenshots below.  I think the button group should probably update its layout in a case like this, which may make this fix workable, but as of this moment it doesn't seem complete.  What's your take?

![image](https://github.com/phetsims/sun/assets/3082831/18d46e98-281d-4237-be7e-28654255ca8a)

![image](https://github.com/phetsims/sun/assets/3082831/49b0e542-9131-47eb-9a36-39eb0799e7c0)

FWIW I didn't continue reviewing the code after seeing this.

**Comment by jbphet** [comment]
I've created a separate issue for addressing the dynamic layout needs of the radio button group, see https://github.com/phetsims/sun/issues/852.

**Comment by jessegreenberg** [comment]
Hmm, good catch. The same can be seen for the mouse/touch areas:

![image](https://github.com/phetsims/sun/assets/6396244/d4863447-fb47-4ef3-9bc1-58632a36d419)

I am not sure how to best handle the layout. I will revert the change, and we can make sure this is fixed after #852.

**Comment by jonathanolson** [comment]
This looks like it was buggy because RectangularRadioButtonGroup was trying to do its own override for focus highlights. Removing this code fixed it. Can you verify?

**Comment by jessegreenberg** [comment]
Awesome, thats great! I tested by running greenhouse-effect with `?supportsDynamicLocale=true&locales=*&keyboardLocaleSwitcher` and changed the strings a few times. All highlights looked correct, even when the button contents had different sizes like in https://github.com/phetsims/sun/issues/851#issuecomment-1686906201.

Thanks for fixing this. Closing.

**Comment by jessegreenberg** [comment]
Reopening, @arouinfar @terracoda and I noticed that the focus highlight no longer surrounds the the `label` Text of the button in the RectangularRadioButtonGroup, if it has one. Ill try to add that back.

**Comment by jessegreenberg** [comment]
@jonathanolson would you mind reviewing this change?

**Comment by jessegreenberg** [comment]
Enough time has passed so I think it is OK to close - sims have gone through QA with this change. Closing.
